addons/l10n_ar/models/account_journal.py

```diff
@@ -164,3 +154,23 @@ def _onchange_set_short_name(self):
         """
         if self.type == 'sale' and self.l10n_ar_afip_pos_number:
             self.code = "%05i" % self.l10n_ar_afip_pos_number
+
+    def write(self, vals):
+        protected_fields = ('type', 'l10n_ar_afip_pos_system', 'l10n_ar_afip_pos_number', 'l10n_latam_use_documents')
+        fields_to_check = [field for field in protected_fields if field in vals]
+
+        if fields_to_check:
+            for journal in self:
+                if (
+                    journal.company_id.account_fiscal_country_id.code != "AR"
+                    or journal.type not in ['sale', 'purchase']
+                    or not self.env['account.move'].search([('journal_id', '=', journal.id), ('posted_before', '=', True)], limit=1)
+                ):
+                    continue
+
+                for field in fields_to_check:
+                    # Wouldn't work if there was a relational field, as we would compare an id with a recordset.
+                    if vals[field] != journal[field]:
+                        raise UserError(_("You can not change the journal's configuration if it already has validated invoices (%s)", journal.name))
```

Comment:
```suggestion
                        raise UserError(_("You can not change %s journal's configuration if it already has validated invoices", journal.name))
```
(just a personal preference though)

Reply:
done :)